### PR TITLE
Add x-placement to tooltip node

### PR DESCRIPTION
**Pull request by atomiks**
No need to think about `.tippy-popper` in themes anymore.

Don't know why I didn't think of doing this sooner 🤦‍♂️

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -21,24 +21,15 @@
...
-      <button class="group">Group</button>
-      <button class="group">Group</button>
-      <button class="group">Group</button>
-
-      <h2>Event delegation</h2>
-      <button class="delegation" data-tippy-content="Event delegation"></button>
-    </div>
+    <div class="container"></div>
+    <!-- prettier-ignore -->
+    <script> </script>
```
Comment (KubaJastrz) [inline demo/index.html]: ?

Comment (atomiks) [reply demo/index.html]: Fixes a Chrome transition bug that's existed for years. I'm using `showOnInit` in the demo and the tooltips transition weirdly on load without this hack. A script tag with an empty space fixes it (so weird, I know)

Comment (atomiks) [reply demo/index.html]: https://stackoverflow.com/a/42969257/9470278

Comment (KubaJastrz) [reply demo/index.html]: Huh, interesting. Please, add a comment about this for the future generations 😉.

**Comment by atomiks** [comment]
Doesn't update the attribute when you do

`instance.set({ placement: new })`

🤔

**Comment by atomiks** [comment]
Looks like this exposed a bug when updating a popperInstance dependency: it should be calling `onCreate` too (reason why it doesn't have to usually is because we're setting `onCreate` and `onUpdate` in `afterPopperPositionUpdates`.

**Comment by atomiks** [comment]
That `afterPopperPositionUpdates` hack only exists due to Popper's lack of callback / promise return type for `.update() / scheduleUpdate()` too... 😞 

I think this should be good now.

**Pull request merged**